### DISCUSSION TITLE
Interactive legend/annotation

### DISCUSSION THREAD

**Discussion by l-mb** [osmandapp/OsmAnd#13294] [Ideas]
Dear all,

I'm extremely happy with OsmAnd as a subscriber, and use it for all my hikes.

There is one detail I'd love to see improved - the map is very detailed, and not only is the legend only online, there's then trying to find which symbol I'm actually looking for.

I'd envision a button/widget/function that gives me an interactive legend, but only for the symbols/types _currently visible_. Or a long tap anywhere on the map and I get a pop-up with all symbols used in the area where I tapped.

Is this an option? I'd be willing to contribute to a bounty.

**Comment by EugeneZmeuk** [comment]
Hello, 
We have the plan about "Map legend" for offline using and we tested some variants.
Your idea is understood for us, and something of your proposal we plan to add.
Now, you can try to download "Map legend" plugin for offline using:
https://github.com/osmandapp/OsmAnd/discussions/12959
https://github.com/osmandapp/OsmAnd/discussions/12959#discussioncomment-1525209
